packages/daemon/src/service/trace-manager.ts

```diff
@@ -0,0 +1,251 @@
+import { Transform, TransformCallback } from 'stream';
+import { EventEmitter } from 'events';
+import { open, close, write } from 'fs-extra';
+import { provide, scope, ScopeEnum } from 'midway';
+import { StringDecoder } from 'string_decoder';
+import { generateId, PipelineStatus, PluginTypeI } from '@pipcook/pipcook-core';
+import Debug from 'debug';
+
+const debug = Debug('daemon.service.tracer');
+
+export type TraceType = 'log' | 'job_status';
+
+/**
+ * base pipcook event, defined the fields `type` and `data`
+ */
+export class TraceEvent {
+  data?: any;
+  constructor(public type: TraceType) {}
+}
+
+/**
+ * pipcook event data type for job status change
+ */
+export class JobStatusChangeEvent extends TraceEvent {
+  data: {
+    jobStatus: PipelineStatus;
+    step?: PluginTypeI;
+    stepAction?: 'start' | 'end';
+    queueLength?: number;
+  };
+  constructor(jobStatus: PipelineStatus, step?: PluginTypeI, stepAction?: 'start' | 'end', queueLength?: number) {
+    super('job_status');
+    this.data = { jobStatus, step, stepAction, queueLength };
+  }
+}
+
+type LogLevel = 'info' | 'warn' | 'error';
+/**
+ * pipcook event data type for log
+ */
+export class LogEvent extends TraceEvent {
+  data: {
+    level: LogLevel;
+    data: string;
+  };
+  constructor(level: LogLevel, data: string) {
+    super('log');
+    this.data = { level, data };
+  }
+}
+
+export type PipcookEvent = JobStatusChangeEvent | LogEvent;
```

Comment:
Ditto, I think the PipcookEvent is not proper for this type, it's used in daemon only.